axum-extra/src/typed_header.rs

```diff
@@ -168,12 +172,55 @@ impl std::error::Error for TypedHeaderRejection {
     }
 }
 
+// copied from https://docs.rs/headers/latest/src/headers/map_ext.rs.html#22-52
+fn typed_insert<H>(headers: &mut http::HeaderMap, header: H)
```

Comment:
Why is it copied, instead of being reused?

Reply:
I originally tried to just depend on headers-core so had to copy this but can just be removed now!